python/ray/serve/_private/application_state.py

```diff
@@ -1089,7 +1116,11 @@ def delete_app(self, name: str) -> None:
             return
         self._application_states[name].delete()
 
-    def deploy_apps(self, name_to_deployment_args: Dict[str, List[Dict]]) -> None:
+    def deploy_apps(
+        self,
+        name_to_deployment_args: Dict[str, List[Dict]],
+        name_to_application_args: Dict[str, ApplicationArgsProto],
```

Comment:
write a unit test in to check when `app_state_manager.deploy_apps` is called with `name_to_application_args` works as expected.